src/meshes/p4est_mesh.jl

```diff
@@ -3,108 +3,115 @@
 # we need to opt-in explicitly.
 # See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
 @muladd begin
-
+#! format: noindent
 
 """
     P4estMesh{NDIMS} <: AbstractMesh{NDIMS}
 
 An unstructured curved mesh based on trees that uses the C library `p4est`
 to manage trees and mesh refinement.
 """
-mutable struct P4estMesh{NDIMS, RealT<:Real, IsParallel, P, Ghost, NDIMSP2, NNODES} <: AbstractMesh{NDIMS}
-  p4est                 ::P # Either Ptr{p4est_t} or Ptr{p8est_t}
-  is_parallel           ::IsParallel
-  ghost                 ::Ghost # Either Ptr{p4est_ghost_t} or Ptr{p8est_ghost_t}
-  # Coordinates at the nodes specified by the tensor product of `nodes` (NDIMS times).
-  # This specifies the geometry interpolation for each tree.
-  tree_node_coordinates ::Array{RealT, NDIMSP2} # [dimension, i, j, k, tree]
-  nodes                 ::SVector{NNODES, RealT}
-  boundary_names        ::Array{Symbol, 2}      # [face direction, tree]
-  current_filename      ::String
-  unsaved_changes       ::Bool
-  p4est_partition_allow_for_coarsening::Bool
-
-  function P4estMesh{NDIMS}(p4est, tree_node_coordinates, nodes, boundary_names,
-                            current_filename, unsaved_changes, p4est_partition_allow_for_coarsening) where NDIMS
-    if NDIMS == 2
-      @assert p4est isa Ptr{p4est_t}
-    elseif NDIMS == 3
-      @assert p4est isa Ptr{p8est_t}
-    end
+mutable struct P4estMesh{NDIMS, RealT <: Real, IsParallel, P, Ghost, NDIMSP2, NNODES} <:
+               AbstractMesh{NDIMS}
+    p4est::P # Either Ptr{p4est_t} or Ptr{p8est_t}
+    is_parallel::IsParallel
+    ghost::Ghost # Either Ptr{p4est_ghost_t} or Ptr{p8est_ghost_t}
+    # Coordinates at the nodes specified by the tensor product of `nodes` (NDIMS times).
+    # This specifies the geometry interpolation for each tree.
+    tree_node_coordinates::Array{RealT, NDIMSP2} # [dimension, i, j, k, tree]
+    nodes::SVector{NNODES, RealT}
+    boundary_names::Array{Symbol, 2}      # [face direction, tree]
+    current_filename::String
+    unsaved_changes::Bool
+    p4est_partition_allow_for_coarsening::Bool
```

Comment:
With the option `align_struct_field = true`, aligning these fields is allowed, but not enforced. It allows both aligning and not aligning.

Reply:
I don't think either way looks *much* better, so both would be fine to me.

Reply:
Then setting this to `true` gives the author of the code the most flexibility. Most structs are perfectly readable without alignment, but some bigger ones like the `P4estMesh` above greatly benefit from alignment IMO.

Reply:
Sounds good to me :+1:

Reply:
Adding `align_struct_field = true` will not help here, since only the first three lines are kept aligned while the remaining lines are as they are right now - the comment in the middle just breaks things 😢